doc/converse/manual.rst

```diff
@@ -2090,6 +2104,56 @@ times affecting performance still apply, This function returns the
 previous (and potentially current, if it was shorter than
 ``newResolution``,) resolution in seconds.
 
+.. _converse_task_queues:
+
+Converse Task Queue
+=====================
+
+In addition to the scheduler queue, Converse provides support for task queues. To enable these queues,
+the flag ``--enable_task_queue`` should be included when Converse is built in SMP mode.
+Converse supports queues managed by the PE's scheduler and those managed by user programs.
+
+To support efficient insertion, deletion, and work-stealing, queues in Converse are implemented according to the work-stealing protocol in the Cilk programming language. That is to say that each thread inserts and removes tasks at the tail of its queue, much like a stack. Threads can also steal work from other threads. A thread
```

Comment:
Here and throughout, I think we should use front/back rather than head/tail, it more clearly denotes a position and it's consistent with the naming scheme of `std::queue`.